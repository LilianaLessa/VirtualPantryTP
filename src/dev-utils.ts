// eslint-disable-next-line import/no-extraneous-dependencies
import { faker } from "@faker-js/faker";
// eslint-disable-next-line import/no-extraneous-dependencies
import { v4 as uuidv4 } from "uuid";
import Product from "./features/products/classes/product.class";
import { IProduct } from "./features/products/interfaces/product.interface";

export const createMockProduct = (): IProduct =>
  new Product(uuidv4(), faker.commerce.productName());
