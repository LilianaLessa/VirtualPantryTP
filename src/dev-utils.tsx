import { faker } from "@faker-js/faker";
// eslint-disable-next-line import/no-extraneous-dependencies
import { v4 as uuidv4 } from "uuid";
import { Text, View } from "react-native";
import React from "react";
import { IProduct } from "./features/products/interfaces/product.interface";
import Product from "./features/products/classes/product.class";
import Pantry from "./features/pantries/classes/pantry.class";
import { IPantry } from "./features/pantries/interfaces/pantry.interface";
import { INotification } from "./features/notification/interfaces/notification.interface";
import IShoppingList from "./features/shoppingList/interfaces/shopping-list.interface";
import ShoppingList from "./features/shoppingList/classes/shopping-list.class";

export const createMockProduct = (): IProduct =>
  new Product(uuidv4(), "", faker.commerce.productName());

export const createMockShoppingLists = (): IShoppingList =>
  new ShoppingList(uuidv4(), faker.word.verb());

export const createMockPantry = (): IPantry =>
  new Pantry(uuidv4(), faker.name.firstName());

export function ScreenPlaceHolder({ route }) {
  const { screenName } = route.params;
  return (
    <View>
      <Text>{screenName} Screen Placeholder</Text>
    </View>
  );
}

export function getStackTraceAsString(e?: Error): string {
  const error = e ?? new Error();
  // console.log(stack);
  return error.stack
    .split("\n")
    .slice(2)
    .map((line) => line.replace(/\s+at\s+/, ""))
    .join("\n");
}
