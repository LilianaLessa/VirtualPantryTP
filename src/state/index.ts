import * as actionCreators from "./action-creators";

export * from "./store";
export * from "./reducers";

/**
 * to solve
 * Export namespace should be first transformed by `@babel/plugin-proposal-export-namespace-from`.
 * @see https://stackoverflow.com/questions/61718768/how-can-i-export-as-namespace-in-babel
 */
export { actionCreators };
