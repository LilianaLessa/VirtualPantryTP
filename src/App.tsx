import * as React from "react";
import { Provider } from "react-redux";
import { NavigationContainer, useNavigation } from "@react-navigation/native";

import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { ActivityIndicator, MD2Colors, Snackbar } from "react-native-paper";
import { SafeAreaProvider } from "react-native-safe-area-context";
import Modal from "react-native-modal";
import { TouchableOpacity, View } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { store } from "./state";
import { useTypedSelector } from "./hooks/useTypedSelector";
import { useActions } from "./hooks/useActions";
import HomeScreen from "./features/home/screens/home.screen";
import { ScreenPlaceHolder } from "./dev-utils";
import {
  EditProductScreen,
  EditProductScreenParams,
} from "./features/products/screens/edit-product.screen";

import { BarCodeScannerContextProvider } from "./services/barCodeScanner/barCodeScanner.context";
import {
  ProductSearchResultScreen,
  ProductSearchResultsScreenParams,
} from "./features/products/screens/product-search-results.screen";
import {
  BarCodeScanScreenRouteName,
  EditProductScreenRouteName,
  ProductScreenRouteName,
  ProductSearchResultScreenRouteName,
} from "./infrastructure/navigation/route-names";
import ProductScreen from "./features/products/screens/product.screen";
import BarCodeScanScreen from "./features/products/screens/barcode-scan.screen";

function LoadingModal() {
  const { fetchingData } = useTypedSelector((state) => state.apiActivity);
  return (
    <Modal isVisible={fetchingData}>
      <ActivityIndicator animating color={MD2Colors.red800} size="large" />
    </Modal>
  );
}

function ErrorSnack() {
  const { errorVisible, errorMessage } = useTypedSelector(
    // eslint-disable-next-line comma-dangle
    (state) => state.messageSnackbar
  );
  const { hideErrorSnack } = useActions();

  return (
    <Snackbar
      style={{
        backgroundColor: "#B22D1D",
      }}
      visible={errorVisible}
      onDismiss={() => hideErrorSnack()}
      action={{
        label: "Dismiss",
        onPress: () => hideErrorSnack(),
      }}
      duration={7000}
    >
      {errorMessage}
    </Snackbar>
  );
}

function InfoSnack() {
  const { infoVisible, infoMessage } = useTypedSelector(
    // eslint-disable-next-line comma-dangle
    (state) => state.messageSnackbar
  );
  const { hideInfoSnack } = useActions();

  return (
    <Snackbar
      visible={infoVisible}
      onDismiss={() => hideInfoSnack()}
      action={{
        label: "Dismiss",
        onPress: () => hideInfoSnack(),
      }}
      duration={3000}
    >
      {infoMessage}
    </Snackbar>
  );
}

function HeaderLeftActions() {
  const navigation = useNavigation();

  return (
    <View>
      <TouchableOpacity
        onPress={() => navigation.navigate("ConfigurationsScreen" as never)}
      >
        <MaterialCommunityIcons name="menu" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
}

function HeaderRightActions() {
  const navigation = useNavigation();

  return (
    <View>
      <TouchableOpacity
        onPress={() => navigation.navigate("NotificationsScreen" as never)}
      >
        <MaterialCommunityIcons name="bell" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
}

type RootStackParamList =
  | { Home: undefined }
  | { Products: undefined }
  | { BarCodeScanScreen: undefined }
  | { NotificationsScreen: { screenName: string } }
  | { ConfigurationsScreen: { screenName: string } }
  | { Pantries: { screenName: string } }
  | { ShoppingLists: { screenName: string } }
  | { Groups: { screenName: string } }
  | ProductSearchResultsScreenParams
  | EditProductScreenParams;

const Stack = createNativeStackNavigator<RootStackParamList>();

function App() {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <Provider store={store}>
          <BarCodeScannerContextProvider>
            <Stack.Navigator initialRouteName="Home">
              <Stack.Screen
                name="Home"
                component={HomeScreen}
                options={{
                  headerLeft: () => <HeaderLeftActions />,
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={ProductScreenRouteName as never}
                component={ProductScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={BarCodeScanScreenRouteName as never}
                component={BarCodeScanScreen}
              />
              <Stack.Screen
                name="NotificationsScreen"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Notifications" }}
              />
              <Stack.Screen
                name="ConfigurationsScreen"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Configurations" }}
              />
              <Stack.Screen
                name="Pantries"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Pantries" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name="ShoppingLists"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Shopping Lists" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name="Groups"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Groups" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={EditProductScreenRouteName as never}
                component={EditProductScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={ProductSearchResultScreenRouteName as never}
                component={ProductSearchResultScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
            </Stack.Navigator>
            <LoadingModal />
            <ErrorSnack />
            <InfoSnack />
          </BarCodeScannerContextProvider>
        </Provider>
      </NavigationContainer>
    </SafeAreaProvider>
  );
}

export default App;
