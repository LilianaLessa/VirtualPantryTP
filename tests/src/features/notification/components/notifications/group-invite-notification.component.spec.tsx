import * as React from "react";
import * as renderer from "react-test-renderer";
import GroupInviteNotificationComponent from "../../../../../../src/features/notification/components/notifications/group-invite-notification.component";
import { createMockNotification } from "../../../../../../src/dev-utils";

describe("App", () => {
  it("snapshot test", () => {
    const notification = createMockNotification();

    const tree = renderer
      .create(<GroupInviteNotificationComponent notification={notification} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
