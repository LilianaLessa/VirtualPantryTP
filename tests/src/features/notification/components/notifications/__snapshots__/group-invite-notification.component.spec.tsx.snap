// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text />
  <View
    className="sc-hLseeU iEikig"
  >
    <Text
      className="sc-eDDNvR dvVKsg"
    >
      You were invite to the group 'undefined'.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderColor": "#000000",
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderColor": "#000000",
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <Text
      className="sc-jTrPJq ikuvwI"
    >
      Tuesday, May 30, 2023, 24:00:00
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
